mediapipe/tasks/ios/vision/gesture_recognizer/sources/MPPGestureRecognizer.h

```diff
@@ -0,0 +1,274 @@
+ * error in initializing the gesture recognizer.
+ */
+- (nullable instancetype)initWithOptions:(MPPGestureRecognizerOptions *)options
+                                   error:(NSError **)error NS_DESIGNATED_INITIALIZER;
+
+/**
+ * Performs gesture recognition on the provided MPPImage using the whole image as region of
+ * interest. Rotation will be applied according to the `orientation` property of the provided
+ * `MPPImage`. Only use this method when the `MPPGestureRecognizer` is created with
+ * `MPPRunningModeImage`.
+ * This method supports gesture recognition of RGBA images. If your `MPPImage` has a source type of
+ * `MPPImageSourceTypePixelBuffer` or `MPPImageSourceTypeSampleBuffer`, the underlying pixel buffer
+ * must have one of the following pixel format types:
+ * 1. kCVPixelFormatType_32BGRA
+ * 2. kCVPixelFormatType_32RGBA
+ *
+ * If your `MPPImage` has a source type of `MPPImageSourceTypeImage` ensure that the color space is
+ * RGB with an Alpha channel.
+ *
+ * @param image The `MPPImage` on which gesture recognition is to be performed.
+ * @param error An optional error parameter populated when there is an error in performing gesture
+ * recognition on the input image.
+ *
+ * @return  An `MPPGestureRecognizerResult` object that contains the hand gesture recognition
+ * results.
+ */
+- (nullable MPPGestureRecognizerResult *)recognizeImage:(MPPImage *)image
+                                                  error:(NSError **)error
+    NS_SWIFT_NAME(recognize(image:));
+
+/**
+ * Performs gesture recognition on the provided `MPPImage` cropped to the specified region of
+ * interest. Rotation will be applied on the cropped image according to the `orientation` property
+ * of the provided `MPPImage`. Only use this method when the `MPPGestureRecognizer` is created with
+ * `MPPRunningModeImage`.
+ *
+ * This method supports gesture recognition of RGBA images. If your `MPPImage` has a source type of
+ * `MPPImageSourceTypePixelBuffer` or `MPPImageSourceTypeSampleBuffer`, the underlying pixel buffer
+ * must have one of the following pixel format types:
+ * 1. kCVPixelFormatType_32BGRA
+ * 2. kCVPixelFormatType_32RGBA
+ *
+ * If your `MPPImage` has a source type of `MPPImageSourceTypeImage` ensure that the color space is
+ * RGB with an Alpha channel.
+ *
+ * @param image The `MPPImage` on which gesture recognition is to be performed.
+ * @param roi A `CGRect` specifying the region of interest within the given `MPPImage`, on which
+ * gesture recognition should be performed.
+ * @param error An optional error parameter populated when there is an error in performing gesture
+ * recognition on the input image.
+ *
+ * @return  An `MPPGestureRecognizerResult` object that contains the hand gesture recognition
+ * results.
+ */
+- (nullable MPPGestureRecognizerResult *)recognizeImage:(MPPImage *)image
+                                       regionOfInterest:(CGRect)roi
```

Comment:
This task does not support region of interest.

Reply:
Sorry my bad. Updated this.